contrib/create_yaml_files.py

```diff
@@ -0,0 +1,274 @@
+                ]
+            else:
+                collisions = [
+                    k
+                    for k, v in tasks_to_new_names_and_yml.items()
+                    if v[0].lower()[:-1] == task_basename.lower()[:-1] and k != old_name
+                ]
+            return collisions
+
+        # store original colissions for rename
+        collisions = _compute_collisions(curr_task, tasks_to_new_names_and_yml)
+        counter = 1
+        while _compute_collisions(curr_task, tasks_to_new_names_and_yml):
+            curr_task = curr_task[:-2] + "-" + str(counter) + curr_task[-2:]
+            counter += 1
+        tasks_to_new_names_and_yml[old_name][0] = curr_task
+        for other in collisions:
+            new_name = tasks_to_new_names_and_yml[other][0]
+            while _compute_collisions(new_name, tasks_to_new_names_and_yml):
+                new_name = new_name[:-2] + "-" + str(counter) + new_name[-2:]
+                counter += 1
+            tasks_to_new_names_and_yml[other][0] = new_name
+
+        task_basename = os.path.basename(curr_task)
+        yml_content = "format_version: '1.0'\n"
+        yml_content += "\n"
+        if CHANGE_FILE_NAME:
+            yml_content += "# old file name: " + os.path.basename(old_name) + "\n"
+        yml_content += "input_files: '" + task_basename + "'\n"
+        yml_content += "\n"
+
+        task_dir = os.path.dirname(curr_task)
+        if not yml_info[1]:
+            yml_content += "properties: []\n"
+        else:
+            yml_content += "properties:\n"
+            for prop, expected in sorted(yml_info[1], key=lambda p: p[0][0]):
+                prop_file = _get_prop(prop[0], prop_dir, task_dir)
+                yml_content += "  - property_file: " + prop_file + "\n"
+                if expected:
+                    yml_content += "    expected_verdict: " + expected + "\n"
+                if prop[1]:
+                    yml_content += "    subproperty: " + prop[1] + "\n"
+                if "unreach-call" in prop_file and expected == "false":
+                    prop_file = _get_prop(
+                        NAME_TO_PROP_AND_SUBPROP["cover-error"][0], prop_dir, task_dir
+                    )
+                    yml_content += "  - property_file: " + prop_file + "\n"
+        yml_file = curr_task[:-2] + ".yml"
+        with open(yml_file, "w+") as outp:
+            outp.write(yml_content)
+
+        if old_name != curr_task:
+            os.rename(old_name, curr_task)
+            if old_name[-1] == "i":
+                # *.i -> *.c
+                if os.path.exists(old_name[:-1] + "c"):
+                    old_c = old_name[:-1] + "c"
+                # *.c.i -> *.c
+                elif os.path.exists(old_name[:-2]):
+                    old_c = old_name[:-2]
+                # ldv-memsafety/memleaks*.i -> ldv-memsafety/memleaks-notpreprocessed/memleaks*.c
+                elif old_name.startswith("ldv-memsafety/memleaks"):
```

Comment:
This could be removed?

Reply:
You're right, that was specific to the conversions in the sv-benchmarks repository. I removed the elif branch.